homeassistant/components/azure_devops/sensor.py

```diff
@@ -0,0 +1,147 @@
+"""Support for Azure DevOps sensors."""
+from datetime import timedelta
+import logging
+from typing import List
+
+from aioazuredevops.builds import DevOpsBuild
+from aioazuredevops.client import DevOpsClient
+import aiohttp
+
+from homeassistant.components.azure_devops import AzureDevOpsDeviceEntity
+from homeassistant.components.azure_devops.const import (
+    CONF_ORG,
+    CONF_PROJECT,
+    DATA_AZURE_DEVOPS_CLIENT,
+    DATA_ORG,
+    DATA_PROJECT,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=300)
+PARALLEL_UPDATES = 4
+
+BUILDS_QUERY = "?queryOrder=queueTimeDescending&maxBuildsPerDefinition=1"
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Azure DevOps sensor based on a config entry."""
+    instance_key = f"{DOMAIN}_{entry.data[CONF_ORG]}_{entry.data[CONF_PROJECT]}"
+    client = hass.data[instance_key][DATA_AZURE_DEVOPS_CLIENT]
+    organization = hass.data[instance_key][DATA_ORG]
+    project = hass.data[instance_key][DATA_PROJECT]
+    sensors = []
+
+    try:
+        builds: List[DevOpsBuild] = await client.get_builds(
+            organization, project, BUILDS_QUERY
+        )
+        for build in builds:
+            sensors.append(
+                AzureDevOpsLatestBuildSensor(client, organization, project, build)
+            )
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise PlatformNotReady from exception
+
+    async_add_entities(sensors, True)
+
+
+class AzureDevOpsSensor(AzureDevOpsDeviceEntity):
+    """Defines a Azure DevOps sensor."""
+
+    def __init__(
+        self,
+        client: DevOpsClient,
+        organization: str,
+        project: str,
+        key: str,
+        name: str,
+        icon: str,
+        measurement: str = "",
+        unit_of_measurement: str = "",
+    ) -> None:
+        """Initialize Azure DevOps sensor."""
+        self._state = None
+        self._attributes = None
+        self._available = False
+        self._unit_of_measurement = unit_of_measurement
+        self.measurement = measurement
+        self.client = client
+        self.organization = organization
+        self.project = project
+        self.key = key
+
+        super().__init__(organization, project, name, icon)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return "_".join([DOMAIN, self.organization, self.key])
```

Comment:
We don't need to include the domain in the unique id. The entity registry is aware of both the integration domain and the platform domain.